lib/bundles/inspec-compliance/api.rb

```diff
@@ -27,15 +27,21 @@ def self.profiles(config)
       # TODO, api should not be dependent on .supported?
       response = Compliance::HTTP.get(url, config['token'], config['insecure'], !config.supported?(:oidc))
       data = response.body
-      if !data.nil?
+      response_code = response.code
+      case response_code
+      when '200'
         profiles = JSON.parse(data)
         # iterate over profiles
         profiles.map do |owner, ps|
           ps.keys.map do |name|
             { org: owner, name: name }
           end
         end.flatten
+      when '401'
+        puts '401 Unauthorized. Please check your token.'
+        []
       else
+        puts response_code, 'An error occured'
```

Comment:
I'm not familiar with the compliance API, but `response.message` will typically be filled with the error message that the server sent back, so you might try something like:

    puts "An unexpected error occurred (HTTP #{response_code}): #{response.message}"